api/src/services/listing.service.ts

```diff
@@ -1753,4 +1758,47 @@ export class ListingService implements OnModuleInit {
       success: true,
     };
   }
+
+  /**
+    runs the job to auto expire lotteries that are passed their due date
+    will call the the cache purge to purge all listings as long as updates had to be made
+  */
+  async expireLotteries(): Promise<SuccessDTO> {
+    if (process.env.LOTTERY_DAYS_TILL_EXPIRY) {
+      this.logger.warn('changeExpiredLotteryStatusCron job running');
+      await this.markCronJobAsStarted(LOTTERY_CRON_JOB_NAME);
+      const expiration_date = new Date();
+      expiration_date.setDate(
+        expiration_date.getDate() -
+          Number(process.env.LOTTERY_DAYS_TILL_EXPIRY),
+      );
+      const res = await this.prisma.listings.updateMany({
+        data: {
+          lotteryStatus: LotteryStatusEnum.expired,
+        },
+        where: {
+          status: ListingsStatusEnum.closed,
+          reviewOrderType: ReviewOrderTypeEnum.lottery,
+          closedAt: {
+            lte: expiration_date,
+          },
+          OR: [
+            {
+              lotteryStatus: {
+                not: LotteryStatusEnum.expired,
+              },
+            },
+            {
+              lotteryStatus: null,
+            },
```

Comment:
The way you have it is probably the correct way to do it, but would lotteryStatus of being null not get caught in `lotteryStatus: {
                not: LotteryStatusEnum.expired,
              }`?

Reply:
Sadly it was actually causing a failure with just the `not` clause